flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java

```diff
@@ -241,6 +244,19 @@ public void putTableSchema(String key, TableSchema schema) {
 				Arrays.asList(WATERMARK_ROWTIME, WATERMARK_STRATEGY_EXPR, WATERMARK_STRATEGY_DATA_TYPE),
 				watermarkValues);
 		}
+
+		if (schema.getPrimaryKey().isPresent()) {
+			final UniqueConstraint uniqueConstraint = schema.getPrimaryKey().get();
+			final List<List<String>> uniqueConstraintValues = new ArrayList<>();
+			uniqueConstraintValues.add(Arrays.asList(
+					uniqueConstraint.getName(),
+					uniqueConstraint.getType().name(),
+					String.join(",", uniqueConstraint.getColumns())));
+			putIndexedFixedProperties(
+					key + '.' + CONSTRAINT_UNIQUE,
+					Arrays.asList(NAME, TYPE, CONSTRAINT_UNIQUE_COLUMNS),
+					uniqueConstraintValues);
+		}
```

Comment:
Because we only support primary key now. I think we can have a dedicate primary key properties, so that we don't need to handle the index. For example:

```java
public static final String PRIMARY_KEY_NAME = "primary-key.name";
public static final String PRIMARY_KEY_COLUMNS = "primary-key.columns";

schema.getPrimaryKey().ifPresent(pk -> {
    putString(key + "." + PRIMARY_KEY_NAME, pk.getName());
    putString(key + "." + PRIMARY_KEY_COLUMNS, String.join(",", pk.getColumns()));
});
```

This is also helpful for users who write yaml: 

```
tables:
  - name: TableNumber1
    type: source-table
    schema:
      primary-key
        name: constraint1
        columns: f1, f2
```